Fix missing pin 28 (VCAP1) in ADS1298_TQFP symbol from Analog_ADC l…

gmsotavio opened the pull request:
This pull request fixes the issue #1242 

I realized that the pin 28 is missing in ADS1298_TQFP symbol from Analog_ADC library.

Before fixing:
![screenshot_2018-12-08_23-38-27](https://user-images.githubusercontent.com/10669626/49692747-282ca900-fb49-11e8-8339-bee8d7f1b592.png)

After fixing:
![screenshot_2018-12-09_00-30-51](https://user-images.githubusercontent.com/10669626/49692788-1d264880-fb4a-11e8-9d48-1549daa10d0a.png)

This pin is mandatory for TQFP package as can be seen in [datasheet](http://www.ti.com/lit/gpn/ads1298).
![screenshot_2018-12-09_00-35-06](https://user-images.githubusercontent.com/10669626/49692804-62e31100-fb4a-11e8-8c53-2c6e3222d498.png)

Regards,
Otavio Augusto Gomes

------------
Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [x] Provide a URL to a datasheet for the symbol(s) you are contributing
- [x] An example screenshot image is very helpful
- [x] Ensure that the associated footprints match the [official footprint library](https://github.com/kicad/kicad-footprints)
- [ ] If there are matching footprint PRs, provide link(s) as appropriate
- [ ] Check the output of the Travis automated check scripts - fix any errors as required

diegoherranz commented:
Thanks for fixing this.
A few comments:
- This may break existing schematics since RESV1 will have moved. However, given it was broken before (VCAP1 was missing), I guess it's acceptable. Librarians?
- NC pins should be moved in. See http://kicad-pcb.org/libraries/klc/S4.6/. This is an issue that existed on this symbol before your modifications, but it would be good to get that fixed at the same time too. COuld you fix that?
- This symbol could use some pin stacking (http://kicad-pcb.org/libraries/klc/S4.3/), but I guess we may want to leave it as in not to break compatibility? Although it could work if done wisely. Again, librarians?

Thanks!

gmsotavio commented:
> Thanks for fixing this.
> A few comments:
> 
>     * This may break existing schematics since RESV1 will have moved. However, given it was broken before (VCAP1 was missing), I guess it's acceptable. Librarians?
> 
>     * NC pins should be moved in. See http://kicad-pcb.org/libraries/klc/S4.6/. This is an issue that existed on this symbol before your modifications, but it would be good to get that fixed at the same time too. COuld you fix that?
> 
>     * This symbol could use some pin stacking (http://kicad-pcb.org/libraries/klc/S4.3/), but I guess we may want to leave it as in not to break compatibility? Although it could work if done wisely. Again, librarians?
> 
> 
> Thanks!

Hi @diegoherranz 

Thanks for your comments. It is my first contribution to Kicad library.

> This may break existing schematics since RESV1 will have moved. However, given it was broken before (VCAP1 was missing), I guess it's acceptable. Librarians?

I guess it's acceptable too.

> NC pins should be moved in. See http://kicad-pcb.org/libraries/klc/S4.6/. This is an issue that existed on this symbol before your modifications, but it would be good to get that fixed at the same time too. COuld you fix that?

Yes, I do.

> This symbol could use some pin stacking (http://kicad-pcb.org/libraries/klc/S4.3/), but I guess we may want to leave it as in not to break compatibility? Although it could work if done wisely. Again, librarians?

Should I fix that, or wait for a librarian guidance?

One question. How I may proceed to edit this pull request with the new fixes? Sorry, I'm not expert in git/GitHub.

Thanks.
Otavio Augusto Gomes

diegoherranz commented:
> Should I fix that, or wait for a librarian guidance? (pin stacking)

Please wait for the input of a librarian. I think it is OK, but I just help what I can here and I'm not a librarian, so I want to leave this decision to them.

> One question. How I may proceed to edit this pull request with the new fixes? Sorry, I'm not expert in git/GitHub.

You can push more commits to your `issue1242` branch (https://github.com/gmsotavio/kicad-symbols/tree/issue1242) and they will automatically show up here.

Thanks!

antoniovazquezblanco commented:
> This may break existing schematics since RESV1 will have moved. However, given it was broken before (VCAP1 was missing), I guess it's acceptable. Librarians?

I would vote acceptable. The last word should come from @poeschlr anyway.

> This symbol could use some pin stacking (http://kicad-pcb.org/libraries/klc/S4.3/), but I guess we may want to leave it as in not to break compatibility? Although it could work if done wisely. Again, librarians?

Given that we are breaking the device, we may as well go wild with this one; otherwise, we would have to push two breaking updates instead of one. Again, @poeschlr ?

Thank you both!

poeschlr commented:
You are right this will break existing schematics. I however see no good way of fixing this symbol without that.

I would further suggest to move the NC pins onto the symbol outline.

---

Stacking pins is again a change. But as @antoniovazquezblanco  said we break old schematics already so lets go wild.

gmsotavio commented:
@poeschlr @antoniovazquezblanco

gmsotavio commented:
I moved the NC pins onto symbol outline and renamed the symbol from ADS1298_TQFP to ADS1298IPAG. The last is the name stated in the data sheet for TQFP variant.

evanshultz commented:
It should be named `ADS1298xPAG` to wildcard the temp range, as required by KLC.

And ordinally I'd wonder if renaming is necessary even in this case. That's total breakage and the user having to chase down the new symbol. But as RESV1 must be grounded, and that would be bad for the VCAP1 pin to be grounded, giving a big obvious "something happened" sign to the user may be best.

Lastly, if you want to add more cleanup to this symbol you could shorten the pins to 100mil, as KLC says, and then enlarge the symbol rectangle so the pin ends are at the same location as before. But this is optional if you don't want to do it.

gmsotavio commented:
@evanshultz I agree, but take into account which ADS1298**I**PAG is the single option available. Must I replace even in this situation? Another similar symbols not seems to care about that.

![screenshot_2018-12-10_23-17-43](https://user-images.githubusercontent.com/10669626/49771954-9c468880-fcd2-11e8-90d6-916261c9edce.png)

> And ordinally I'd wonder if renaming is necessary even in this case. That's total breakage and the user having to chase down the new symbol.

I agree, It will break old design, but as  @poeschlr said, "...we break old schematics already so lets go wild." Moreover, I can't believe that some Kicad user who used that symbol in a past project has succeeded, because without VCAP1 pin, the ADC simple doesn't works. I intend to add all variants of this symbol to Kicad library, then I need to fix this symbol correctly.

>Lastly, if you want to add more cleanup to this symbol you could shorten the pins to 100mil, as KLC says, and then enlarge the symbol rectangle so the pin ends are at the same location as before. But this is optional if you don't want to do it.

Shorten pins to 100mil looks good, but why enlarge the rectangle symbol could add cleanup to it?

Thanks!

evanshultz commented:
Sorry, out of time today. I'll answer tomorrow.

evanshultz commented:
KLC does not make any allowance for not wildcarding temp. See http://kicad-pcb.org/libraries/klc/S2.2/. More temp variants may be added in the future and, even if some symbols were created before this rule, this is being wildcarded in new symbols.

About breaking the symbols, since we are breaking this one, my statement was to point out if the symbol is removed from the schematic it will be clear some change happened and the old symbol had an issue. Changing the name of the symbol, as you've done, is fine.

If the pins are shortened to 100mil, as KLC states for this 64 pin symbol (pin numbers are all <= 2 characters) at http://kicad-pcb.org/libraries/klc/S4.1/, keeping pin placement the same will require enlarging the symbol. If the pins are simply shortened then all wires on existing schematics will be broken, but perhaps that has the added benefit of alerting any users that the symbol needs some attention (and as we know if somebody is using this symbol and they don't know there's a problem they are daft or haven't yet sent out the PCB).

A couple other things as I look at this symbol in depth since it's getting changed anyway:
1. All pin styles can be Line. We aren't using Clock and other styles.
1. Pin name position offset should be 20mil.
1. All DGND, AVSS, AVDD, and DVDD pins can be stacked. Pages 96-97 show that not each individual pin, but power pins grouped by name, needs capacitance.
1. VREFP should probably be Input type (like VREFN).
1. VREFP and VREFN could be spread apart to let a cap fit in there.
1. RLDINV might be best as an Input type. From the perspective of the RLD circuitry, from which it is named, it's an input.
1. The VCAPx pins should to be on the left (analog) side.
1. It might be nice if RLDINV and RLDOUT were spaced apart a bit for the feedback parts to fit there.
1. I would group DAISY_IN with the other SPI signals, putting it at the top and DOUT at the bottom of the group to allow the easiest signal routing for devices that are daisy-chained.
1. The NC pins can be placed between other pins instead of consuming two consecutive "slots" for pins.
1. Shrink the symbol vertically and horizontally once that is all done.

Note that none of this is required. But if I were reviewing this symbol as a new contribution, these are the suggestions I'd make.

antoniovazquezblanco commented:
> ...keeping pin placement the same will require enlarging the symbol...

Pin placement has already changed. We agreed to break compatibility regarding pin positioning already.

poeschlr commented:
Regarding wildcarding temperature specifiers: Think of it that way. If the manufacturer ever adds another option we do not need to update the symbol. (assuming the option is replaced with the wildcard.)
Meaning the lib is more future proof when using wildcards.

evanshultz commented:
@gmsotavio 
Will you be coming back to finish this PR soon?
